h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/rabit/util/LinkMap.java

```diff
@@ -77,13 +77,19 @@ public AssignedRank assignRank(int rank) {
     }
 
     private List<Integer> getNeighbours(int rank) {
```

Comment:
Does the Scala implementation have unit tests we can re-use?

Reply:
unfortunately no - they have one test suite but it tests their use of Akka for communication (which we don't have, I went with the simpler Java NIO loop based on Python tracker - not optimal but easier to implement), nothing for the remaining logic.